core/power/impl/power_table_hamt.cpp

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "power/impl/power_table_hamt.hpp"
+
+#include "power/power_table_error.hpp"
+#include "primitives/address/address_codec.hpp"
+
+using fc::power::Power;
+using fc::power::PowerTableError;
+using fc::power::PowerTableHamt;
+using fc::primitives::address::Address;
+using fc::primitives::address::decodeFromByteString;
+using fc::primitives::address::encodeToByteString;
+using fc::storage::hamt::HamtError;
+using fc::storage::hamt::Value;
+
+PowerTableHamt::PowerTableHamt(Hamt hamt) : power_table_{std::move(hamt)} {}
+
+fc::outcome::result<Power> PowerTableHamt::getMinerPower(
+    const Address &address) const {
+  auto result = power_table_.getCbor<Power>(encodeToByteString(address));
+  if (!result && result.error() == HamtError::NOT_FOUND) {
+    return outcome::failure(PowerTableError::NO_SUCH_MINER);
+  }
+  return result;
+}
+
+fc::outcome::result<void> PowerTableHamt::setMinerPower(const Address &address,
+                                                        Power power_amount) {
+  if (power_amount < 0) return PowerTableError::NEGATIVE_POWER;
+  return power_table_.setCbor(encodeToByteString(address), power_amount);
+}
+
+/** @copydoc PowerTable::getMinerPower() */
```

Comment:
```suggestion
/** @copydoc PowerTable::removeMiner() */
```
Also others functions

Reply:
we need docs in headers, not in sources

Reply:
move descriptions to headers

Reply:
fixed